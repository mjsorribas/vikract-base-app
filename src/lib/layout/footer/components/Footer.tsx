import {
  Box,
  Container,
  SimpleGrid,
  Stack,
  Divider,
  useColorModeValue,
  GridItem,
} from "@chakra-ui/react";

import Logo from "../../../components/logo/Logo";

import Copyright from "./copyright/Copyright";
import Sitemap from "./sitemap/Sitemap";
import SocialLinks from "./sociallinks/SocialLinks";
import TermsAndConditions from "./termsandconditions/TermsAndConditions";

export default function Footer() {
  return (
    <Box
      bg={useColorModeValue("gray.50", "gray.900")}
      color={useColorModeValue("gray.700", "gray.200")}
      alignContent="center"
      w="100%"
    >
      <Container as={Stack} maxW="full" py={10}>
        <SimpleGrid
          templateColumns={{ sm: "repeat(1, 1fr)", md: "repeat(4, 1fr)" }}
          gap={10}
          spacing={8}
        >
          <GridItem colSpan={{ sm: 1, md: 2 }}>
            <Logo />
          </GridItem>
          <GridItem colSpan={1}>
            <Sitemap />
          </GridItem>
          <GridItem colSpan={1}>
            <SocialLinks />
          </GridItem>
        </SimpleGrid>
        <Divider orientation="horizontal" />
        <SimpleGrid
          templateColumns={{ sm: "repeat(1, 1fr)", md: "repeat(2, 1fr)" }}
          spacing={2}
        >
          <GridItem colSpan={1}>
            <Copyright />
          </GridItem>
          <GridItem colSpan={1}>
            <TermsAndConditions />
          </GridItem>
        </SimpleGrid>
      </Container>
    </Box>
  );
}
