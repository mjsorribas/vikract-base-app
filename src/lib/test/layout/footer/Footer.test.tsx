import { ChakraProvider } from "@chakra-ui/react"; // Importa ChakraProvider
import renderer from "react-test-renderer";
import { expect, test } from "vitest";

import Footer from "lib/layout/footer/Footer";

const toJson = (component: renderer.ReactTestRenderer) => {
  const result = component.toJSON();
  expect(result).toBeDefined();
  return result as renderer.ReactTestRendererJSON;
};

test("Footer", () => {
  const component = renderer.create(
    <ChakraProvider>
      <Footer />
    </ChakraProvider>
  );
  const tree = toJson(component);
  expect(tree).toMatchSnapshot();
});
