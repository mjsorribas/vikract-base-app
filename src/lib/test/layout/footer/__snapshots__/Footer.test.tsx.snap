// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Footer 1`] = `null`;
